.circleci/config.yml

```diff
@@ -1,64 +1,156 @@
+      - run: *install-rebar
+      - restore_cache: *restore-mix
+      - restore_cache:  *restore-build
       - run: mix deps.get
       - run: mix compile --warnings-as-errors --force
       - run: mix format --check-formatted --dry-run
-      - save_cache:
-          key: -{{ .Environment.CACHE_VERSION }}-mix-cache-{{ checksum ".tool-versions" }}-{{ .Branch }}-{{ checksum "mix.lock" }}
-          paths: "deps"
-      - save_cache:
-          key: -{{ .Environment.CACHE_VERSION }}-mix-cache-{{ .Branch }}-{{ checksum "mix.lock" }}
-          paths: "deps"
-      - save_cache:
-          key: -{{ .Environment.CACHE_VERSION }}-mix-cache-{{ .Branch }}
-          paths: "deps"
-      - save_cache:
-          key: -{{ .Environment.CACHE_VERSION }}-mix-cache
-          paths: "deps"
-      - save_cache:
-          key: -{{ .Environment.CACHE_VERSION }}-build-cache-{{ .Branch }}
-          paths: "_build"
-      - save_cache:
-          key: -{{ .Environment.CACHE_VERSION }}-build-cache
-          paths: "_build"
+      - save_cache: *cache-mix
+      - save_cache: *cache-build
+      - persist_to_workspace:
+          root: .
+          paths: .
 
+  test:
+    parallelism: 1
+    working_directory: ~/app
+    docker:
+      - *elixir
+      - *postgres
+    steps:
+      - attach_workspace:
+          at: ~/app
+      - run: *install-rebar
       - run:
           name: Wait for DB
           command: dockerize -wait tcp://localhost:5432 -timeout 1m
-
-      - run: mix do compile --warnings-as-errors, coveralls.json
-
+      - run: MIX_ENV=test mix coveralls.json
       - run: curl -s https://codecov.io/bash | bash
-
       - store_test_results:
           path: _build/test/lib/
+          paths: .
+
+  build_release: &build-release-job
+    parallelism: 1
+    working_directory: ~/app
+    docker:
+      - *elixir-release
+    steps:
+      - attach_workspace:
+          at: ~/app
+      - run: *install-rebar
+      - run: MIX_ENV=prod mix release
+      - run: mv _build/prod/rel/boilerplate _build/prod/rel/$EB_ENV
+      - persist_to_workspace:
+          root: _build/prod
+          paths:
+            - rel
+  build_release_staging:
+    <<: *build-release-job
+    environment: *env-staging
+  build_release_production:
+    <<: *build-release-job
+    environment: *env-production
+
+  deploy: &deploy-job
+    parallelism: 1
+    working_directory: ~/app
+    docker:
+      - *awsebcli
+    steps:
+      - attach_workspace:
+          at: ~/app
+      - run: mv -f Dockerfile.aws Dockerfile
+      - run: mv rel/$EB_ENV rel/boilerplate
+      - run: eb use $EB_ENV
+      - run: eb deploy
+  deploy_staging:
+    <<: *deploy-job
+    environment: *env-staging
+  deploy_production:
+    <<: *deploy-job
+    environment: *env-production
+
+workflows:
+  commit:
+    jobs:
+      - build
+      - build_release_production:
+          context: boilerplate-production
+          requires:
+            - build
```

Comment:
maybe is a good idea require `test` here as well, so we don't deploy if tests aren't finished...

Reply:
`test` is required on approval steps which block the ones that actually deploy a release to eb.